okhttp-tests/src/test/java/okhttp3/internal/framed/HttpOverSpdyTest.java

```diff
@@ -582,6 +583,99 @@ protected HttpOverSpdyTest(Protocol protocol) {
     assertEquals(0, server.takeRequest().getSequenceNumber());
   }
 
+  @Test public void recoverFromOneRefusedStreamReusesConnection() throws Exception {
+    server.enqueue(new MockResponse()
+        .setSocketPolicy(SocketPolicy.RESET_STREAM_AT_START)
+        .setHttp2ErrorCode(ErrorCode.REFUSED_STREAM.httpCode));
+    server.enqueue(new MockResponse()
+        .setBody("abc"));
+
+    Call call = client.newCall(new Request.Builder()
```

Comment:
What happens if you use the `SingleInetAddressDns` here? I was testing with nginx 1.9.15 and I noticed the refuse stream stack trace. I think it's because I was using the IP address directly, but still investigating.